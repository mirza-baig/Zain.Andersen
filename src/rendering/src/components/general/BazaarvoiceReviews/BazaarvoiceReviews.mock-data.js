const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'Iframe',
  },
  fields: {
    "sectionId": {
      "value": ""
    },
    "componentSpacing": null,
    "headlineLevel": null,
    "headlineText": {
      "value": "Bazaarvoice Headline"
    },
    "productItem": {
      "id": "eb76424a-cd01-4826-a6c9-022cc88b95b3",
      "url": "/Data/Product/Products/Casement-Windows/100-Series",
      "name": "100 Series",
      "displayName": "100 Series",
      "fields": {
        "exteriorSwatchCollection": {
          "id": "9b3a3027-c1e5-4e46-8825-0eaf8bb13287",
          "url": "/Data/Elements/Swatches/Collection/Doors-Interior",
          "name": "Doors Interior",
          "displayName": "Doors Interior",
          "fields": {
            "swatchCollectionName": {
              "value": "Doors Exterior"
            },
            "swatches": [
              {
                "id": "92934436-7d09-4463-a661-7dd082712dd8",
                "url": "/Data/Elements/Swatches/Swatch/White",
                "name": "White",
                "displayName": "White",
                "fields": {
                  "swatchName": {
                    "value": "White"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/white-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=5367F261E6FF3E8CD80096C619033F70",
                      "alt": "white-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "f4764f8a-2bdb-46be-93ba-93405ce7c167",
                "url": "/Data/Elements/Swatches/Swatch/Terratone",
                "name": "Terratone",
                "displayName": "Terratone",
                "fields": {
                  "swatchName": {
                    "value": "Terratone"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/terratone-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=4270D1F8BCD0A71CF729CECA23633C0E",
                      "alt": "terratone-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "a564fcb1-4b07-4cd6-8e24-c43d21a63a5b",
                "url": "/Data/Elements/Swatches/Swatch/Sandstone",
                "name": "Sandstone",
                "displayName": "Sandstone",
                "fields": {
                  "swatchName": {
                    "value": "Sandstone"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/sandtone-exterior-color-swatch.png?h=77&iar=0&w=77&hash=DE3C8BC480C3FA9679268AD37FBECDF0",
                      "alt": "sandtone-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "fc8a2adf-8bab-439a-9959-465f21b030de",
                "url": "/Data/Elements/Swatches/Swatch/Dark-Bronze",
                "name": "Dark Bronze",
                "displayName": "Dark Bronze",
                "fields": {
                  "swatchName": {
                    "value": "White"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/dark-bronze-exterior-color-swatch.png?h=77&iar=0&w=77&hash=5601A1FC31A09F9034E2194A03F55A8C",
                      "alt": "dark-bronze-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "39a4fc7a-dd5f-4a80-882e-2be6e06ecd0e",
                "url": "/Data/Elements/Swatches/Swatch/Cocoa-Bean",
                "name": "Cocoa Bean",
                "displayName": "Cocoa Bean",
                "fields": {
                  "swatchName": {
                    "value": "Cocoa Bean"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/cocoa-bean-exterior-color-swatch.png?h=77&iar=0&w=77&hash=47CA54DCBF4366663A3FD4F2617B58DE",
                      "alt": "cocoa-bean-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "56c5510d-b7ce-4df4-a78e-3643f6a1238a",
                "url": "/Data/Elements/Swatches/Swatch/Black",
                "name": "Black",
                "displayName": "Black",
                "fields": {
                  "swatchName": {
                    "value": "Black"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/black-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=2C4CC0B46040C4796AA45B2766275179",
                      "alt": "black-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              }
            ],
            "swatchCollectionDescription": {
              "value": ""
            },
            "swatchCollectionFooterCopy": {
              "value": ""
            }
          },
          "templateId": "c7e82176-d156-410f-b728-d4c465148578",
          "templateName": "Swatch Collection"
        },
        "interiorSwatchCollection": {
          "id": "1fee6bae-f4ef-423c-956e-2aa1ddbf4ea6",
          "url": "/Data/Elements/Swatches/Collection/Doors-Exterior",
          "name": "Doors Exterior",
          "displayName": "Doors Exterior",
          "fields": {
            "swatchCollectionName": {
              "value": "Doors Exterior"
            },
            "swatches": [
              {
                "id": "92934436-7d09-4463-a661-7dd082712dd8",
                "url": "/Data/Elements/Swatches/Swatch/White",
                "name": "White",
                "displayName": "White",
                "fields": {
                  "swatchName": {
                    "value": "White"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/white-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=5367F261E6FF3E8CD80096C619033F70",
                      "alt": "white-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "a564fcb1-4b07-4cd6-8e24-c43d21a63a5b",
                "url": "/Data/Elements/Swatches/Swatch/Sandstone",
                "name": "Sandstone",
                "displayName": "Sandstone",
                "fields": {
                  "swatchName": {
                    "value": "Sandstone"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/sandtone-exterior-color-swatch.png?h=77&iar=0&w=77&hash=DE3C8BC480C3FA9679268AD37FBECDF0",
                      "alt": "sandtone-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "f4764f8a-2bdb-46be-93ba-93405ce7c167",
                "url": "/Data/Elements/Swatches/Swatch/Terratone",
                "name": "Terratone",
                "displayName": "Terratone",
                "fields": {
                  "swatchName": {
                    "value": "Terratone"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/terratone-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=4270D1F8BCD0A71CF729CECA23633C0E",
                      "alt": "terratone-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "39a4fc7a-dd5f-4a80-882e-2be6e06ecd0e",
                "url": "/Data/Elements/Swatches/Swatch/Cocoa-Bean",
                "name": "Cocoa Bean",
                "displayName": "Cocoa Bean",
                "fields": {
                  "swatchName": {
                    "value": "Cocoa Bean"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/cocoa-bean-exterior-color-swatch.png?h=77&iar=0&w=77&hash=47CA54DCBF4366663A3FD4F2617B58DE",
                      "alt": "cocoa-bean-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "56c5510d-b7ce-4df4-a78e-3643f6a1238a",
                "url": "/Data/Elements/Swatches/Swatch/Black",
                "name": "Black",
                "displayName": "Black",
                "fields": {
                  "swatchName": {
                    "value": "Black"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/black-exterior-80x80-(1).jpg?h=80&iar=0&w=80&hash=2C4CC0B46040C4796AA45B2766275179",
                      "alt": "black-exterior-80x80 (1)",
                      "width": "80",
                      "height": "80"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              },
              {
                "id": "fc8a2adf-8bab-439a-9959-465f21b030de",
                "url": "/Data/Elements/Swatches/Swatch/Dark-Bronze",
                "name": "Dark Bronze",
                "displayName": "Dark Bronze",
                "fields": {
                  "swatchName": {
                    "value": "White"
                  },
                  "swatchDescription": {
                    "value": ""
                  },
                  "swatchImage": {
                    "value": {
                      "src": "/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Swatches/dark-bronze-exterior-color-swatch.png?h=77&iar=0&w=77&hash=5601A1FC31A09F9034E2194A03F55A8C",
                      "alt": "dark-bronze-exterior-color-swatch",
                      "width": "77",
                      "height": "77"
                    }
                  }
                },
                "templateId": "863dcc2e-cca5-4236-b10f-8b4872e1c47c",
                "templateName": "Swatch"
              }
            ],
            "swatchCollectionDescription": {
              "value": ""
            },
            "swatchCollectionFooterCopy": {
              "value": ""
            }
          },
          "templateId": "c7e82176-d156-410f-b728-d4c465148578",
          "templateName": "Swatch Collection"
        },
        "productSeries": {
          "id": "ed4c1492-6ff4-4363-a368-c87821a2d411",
          "url": "/Data/Product/Series/100-SERIES",
          "name": "100 SERIES",
          "displayName": "100 SERIES",
          "fields": {
            "seriesDescription": {
              "value": ""
            },
            "seriesImage": {
              "value": {}
            },
            "seriesSubtitle": {
              "value": ""
            },
            "seriesTitle": {
              "value": "100 SERIES"
            }
          },
          "templateId": "a9278813-cfb4-475e-b618-b447475adc18",
          "templateName": "Series"
        },
        "productName": {
          "value": "100 SERIES CASEMENT WINDOW"
        },
        "priceLevel": {
          "id": "f25bc07f-3475-4b1e-8e55-93db370f85b2",
          "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Price-Range-Icons/5",
          "name": "5",
          "displayName": "5",
          "fields": {
            "Value": {
              "value": "5"
            }
          },
          "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
          "templateName": "Enum"
        },
        "productId": {
          "value": "123"
        },
        "productDescription": {
          "value": "<ul>\n    <li>Rigid vinyl exterior helps protect against water damage</li>\n    <li>\n    Rich natural pine or prefinished White, Dark Bronze or Black interiors</li>\n    <li>\n    Our best-selling awning window</li>\n</ul>"
        },
        "featuredExteriorColors": [],
        "featuredInteriorColors": [],
        "bazaarvoiceProductId": {
          "value": "100-CAS"
        },
        "sectionId": {
          "value": ""
        },
        "componentSpacing": null,
        "eventName": {
          "value": ""
        },
        "eventType": {
          "value": ""
        },
        "eventZone": {
          "value": ""
        }
      },
      "templateId": "afe4a298-0fb4-4d78-a699-e96b98ce94de",
      "templateName": "Product"
    },
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
