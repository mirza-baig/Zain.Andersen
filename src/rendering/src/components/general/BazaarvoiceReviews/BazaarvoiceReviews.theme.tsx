// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const BazaarvoiceReviewsTheme: ThemeFile = {
  aw: {
    classes: {
      wrapperClass: 'flex flex-col col-span-12 border-solid border-gray border-y py-s lg:py-ml',
      headline: 'col-span-12 text-sm-m lg:text-m leading-[1.25] font-heavy',
      accordionToggleContainer:
        'group flex flex-row grow w-full self-justify relative hover:cursor-pointer',
      accordionHeadline: 'w-full',
      readMore: `lg:inline mr-m mb-xxs lg:mb-0 text-xs font-heavy font-sans group-hover:underline`,
      accordionRatingContainer: `inline-block mr-m align-middle`,
      accordionToggleIndicator: `flex items-center justify-center inline absolute right-0 top-[calc(50%_-_1rem)] h-[2rem] w-[2rem] rounded-full border-2 border-primary group-hover:bg-primary group-hover:text-white`,
      iconClass: `inline`,
      contentOpen: '',
      contentClosed: 'h-0 overflow-hidden',
    },
  },
  rba: {
    classes: {
      wrapperClass: 'flex flex-col col-span-12 border-solid border-gray border-y py-s lg:py-l',
      headline: 'col-span-12 text-s lg:text-l font-extra-light text-theme-text',
      accordionToggleContainer:
        'group flex flex-row grow w-full self-justify relative hover:cursor-pointer',
      accordionHeadline: 'w-full',
      readMore: `lg:inline mr-m mb-xxs lg:mb-0 text-xs font-heavy group-hover:underline`,
      accordionRatingContainer: `inline-block mr-m align-middle`,
      accordionToggleIndicator: `flex items-center justify-center inline absolute right-0 top-[calc(50%_-_1rem)] h-[2rem] w-[2rem] rounded-full border-2 border-primary group-hover:bg-primary group-hover:text-white`,
      iconClass: `inline`,
      contentOpen: '',
      contentClosed: 'h-0 overflow-hidden',
    },
  },
};
