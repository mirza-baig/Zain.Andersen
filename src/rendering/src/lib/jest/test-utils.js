// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
