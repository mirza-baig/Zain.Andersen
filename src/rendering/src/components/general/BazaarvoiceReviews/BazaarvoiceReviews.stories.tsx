// Global
import { Story, Meta } from '@storybook/react';
// Local
import BazaarvoiceReviews, { BazaarvoiceReviewsProps } from './BazaarvoiceReviews';
import defaultData from './BazaarvoiceReviews.mock-data';

export default {
  title: 'General/BazaarvoiceReviews',
  component: BazaarvoiceReviews,
} as Meta;
const Template: Story<BazaarvoiceReviewsProps> = (props) => (
  <>
    <script
      async
      src="https://apps.bazaarvoice.com/deployments/andersenwindows/main_site/staging/en_US/bv.js"
    ></script>
    <BazaarvoiceReviews {...props} />
  </>
);

export const Default = Template.bind({});
Default.args = defaultData;
