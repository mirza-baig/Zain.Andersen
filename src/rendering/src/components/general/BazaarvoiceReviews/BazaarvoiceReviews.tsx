// Global
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { useTheme } from 'lib/context/ThemeContext';
import { useBVScript } from 'lib/utils/use-bv-script';
import { useEffect, useState } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { BazaarvoiceReviewsTheme } from './BazaarvoiceReviews.theme';

export type BazaarvoiceReviewsProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.General.BazaarvoiceReviews.BazaarvoiceReviews;
const BazaarvoiceReviews = (props: BazaarvoiceReviewsProps) => {
  const { themeData } = useTheme(BazaarvoiceReviewsTheme);
  const productItem = props?.fields?.productItem;
  const bazaarvoiceProductId = productItem?.fields?.bazaarvoiceProductId?.value;
  const componentId = 'bazaarvoiceReview-' + bazaarvoiceProductId;
  const [isOpen, setOpen] = useState(false);

  // Add the bazaarvoice script
  useBVScript();

  const [hasMounted, setHasMounted] = useState(false);
  useEffect(() => {
    setHasMounted(true);
  }, []);

  if (!hasMounted) {
    return null;
  }

  return (
    <Component variant="lg" dataComponent="general/bazaarvoicereviews" {...props}>
      <Headline classes={themeData.classes.headline} {...props} />
      <div className={themeData.classes.wrapperClass}>
        <a
          className={themeData.classes.accordionToggleContainer}
          data-anchor-name={'#' + componentId}
        >
          <h3 className={themeData.classes.accordionHeadline} onClick={() => setOpen(!isOpen)}>
            <div className={themeData.classes.readMore}>Read Reviews</div>
            <div
              className={themeData.classes.accordionRatingContainer}
              data-bv-show="rating_summary"
              data-bv-product-id={bazaarvoiceProductId}
            ></div>
            <div className={themeData.classes.accordionToggleIndicator}>
              <SvgIcon
                icon={isOpen ? 'minus' : 'plus'}
                size="18"
                className={themeData.classes.iconClass}
              />
            </div>
          </h3>
        </a>
        <div
          className={`content-container ${
            isOpen ? themeData.classes.contentOpen : themeData.classes.contentClosed
          }`}
          id={componentId}
        >
          <div className="content">
            <div data-bv-show="reviews" data-bv-product-id={bazaarvoiceProductId}></div>
          </div>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<BazaarvoiceReviewsProps>(BazaarvoiceReviews);
