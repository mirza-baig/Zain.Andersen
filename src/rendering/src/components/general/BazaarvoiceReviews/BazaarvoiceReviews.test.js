// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import BazaarvoiceReviews from './BazaarvoiceReviews';
import defaultData from './BazaarvoiceReviews.mock-data';

it('renders correctly', async () => {
  const component = snapshot(BazaarvoiceReviews, { componentProps: defaultData });
  await hasDataComponent(component, 'general/bazaarvoicereviews');
});
