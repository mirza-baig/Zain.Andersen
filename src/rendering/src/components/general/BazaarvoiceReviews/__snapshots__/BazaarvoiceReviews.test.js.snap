// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/bazaarvoicereviews"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="items-top font-sans [&_a:hover]:underline col-span-12 text-sm-m lg:text-m leading-[1.25] font-heavy"
      >
        <h2>
          Bazaarvoice Headline
        </h2>
      </div>
      <div
        class="flex flex-col col-span-12 border-solid border-gray border-y py-s lg:py-ml"
      >
        <a
          class="group flex flex-row grow w-full self-justify relative hover:cursor-pointer"
          data-anchor-name="#bazaarvoiceReview-100-CAS"
        >
          <h3
            class="w-full"
          >
            <div
              class="lg:inline mr-m mb-xxs lg:mb-0 text-xs font-heavy font-sans group-hover:underline"
            >
              Read Reviews
            </div>
            <div
              class="inline-block mr-m align-middle"
              data-bv-product-id="100-CAS"
              data-bv-show="rating_summary"
            />
            <div
              class="flex items-center justify-center inline absolute right-0 top-[calc(50%_-_1rem)] h-[2rem] w-[2rem] rounded-full border-2 border-primary group-hover:bg-primary group-hover:text-white"
            >
              <span
                class="inline"
              />
            </div>
          </h3>
        </a>
        <div
          class="content-container h-0 overflow-hidden"
          id="bazaarvoiceReview-100-CAS"
        >
          <div
            class="content"
          >
            <div
              data-bv-product-id="100-CAS"
              data-bv-show="reviews"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
